### ISSUE TITLE
train model with imagenet dataset

### ISSUE THREAD

**Issue by zjupx** [YangHM/Convolutional-Prototype-Learning#1]
thanks for this great job.  your paper just report the minist\cifar scores.  do you have train model for imagenet or other largescale dataset?  can you share more experience for this largescale image classification task?  thank you very much.

**Comment by YangHM** [comment]
We don't train the model on ImageNet but train it on another large scale dataset: CASIA-OLHWDB, which has 3755 classes, 2697673 training samples and 224589 test samples. It seems the GCPL model can perform a little better than the traditional soft-max based CNN on this dataset. On large scale dataset, the initialization of the prototypes is important. From our experience, we first initialize the model, then compute the mean (average) for each class with the training data, then assign the mean to the prototype of the corresponding class for initialization.

**Comment by YangHM** [comment]
> thanks for this great job. your paper just report the minist\cifar scores. do you have train model for imagenet or other largescale dataset? can you share more experience for this largescale image classification task? thank you very much.

**Comment by YangHM** [comment]
> thanks for this great job. your paper just report the minist\cifar scores. do you have train model for imagenet or other largescale dataset? can you share more experience for this largescale image classification task? thank you very much.

We don't train the model on ImageNet but train it on another large scale dataset: CASIA-OLHWDB, which has 3755 classes, 2697673 training samples and 224589 test samples. It seems the GCPL model can perform a little better than the traditional soft-max based CNN on this dataset. On large scale dataset, the initialization of the prototypes is important. From our experience, we first initialize the model, then compute the mean (average) for each class with the training data, then assign the mean to the prototype of the corresponding class for initialization.

**Comment by deyituo** [comment]
Have you compared with other softmax based loss such as cosface ?
